providers/sample/src/java/org/sakaiproject/provider/user/SampleCandidateDetailProvider.java

```diff
@@ -201,7 +202,77 @@ public Optional<List<String>> getAdditionalNotes(User user, Site site){
+		} catch(Exception e) {
+			log.warn("Error on isAdditionalNotesEnabled (sample) ", e);
+		}
+		return false;
+	}
+
+	public Optional<List<String>> getSpecialNeeds(User user, Site site){
+		if(site == null) {
+			log.error("getSpecialNeeds: Null site.");
+			return Optional.empty();
+		}
+
+		try {
+			//check if special needs info is enabled (system-wide or site-based)
+			if (user != null && isSpecialNeedsEnabled(site)) {
+				if (user.getProperties() != null && user.getProperties().getPropertyList(USER_PROP_SPECIAL_NEEDS) != null) {
+					log.debug("Showing special needs info for user {}", user.getId());
+					List<String> ret = new ArrayList<String>();
+					for (String s : user.getProperties().getPropertyList(USER_PROP_SPECIAL_NEEDS)) {
+						//this property is encrypted, so we need to decrypt it
+						if(StringUtils.isNotBlank(s) && StringUtils.isNotBlank(encryptionUtilities.decrypt(s))) {
+							ret.add(encryptionUtilities.decrypt(s));
+						}
+					}
+					return Optional.ofNullable(ret);
+				} else {
+					List<String> ret = new ArrayList<String>();
+					int hashInt = user.getId().hashCode();
+					if (hashInt % 10 == 2) {
+						log.debug("Not generating random special needs infos for user {}", user.getId());
+						return Optional.empty();
+					} else {
+						log.debug("Generating random special needs infos for user {}", user.getId());
+						String[] sampleSpecialNeeds = {
+							"Anticipate the material to work on, for example upload the teaching material ahead of time.",
+							"Don't overload information pages.",
+							"At the beginning of the class, the teacher should indicate how he will organize the session and summarize the most important ideas at the end of the session.	",
+							"Respect if student does not want to read out loud in front of classmates.",
+							"Less demanding spelling.",
+							"More time to take the exams.",
+							"Brief problem statements divided into parts."
+						};
+						ret.add(sampleSpecialNeeds[random.nextInt(sampleSpecialNeeds.length)]);
+						ret.add(sampleSpecialNeeds[random.nextInt(sampleSpecialNeeds.length)]);
+						if (hashInt % 10 == 7) {
+							log.debug("Generating more random special needs infos for user {}", user.getId());
+							ret.add(sampleSpecialNeeds[random.nextInt(sampleSpecialNeeds.length)]);
+						}
+						return Optional.ofNullable(ret);
+					}
+				}
+			}
+		} catch (Exception e) {
+			log.error("Error getting sample special needs info for {}", ((user != null) ? user.getId() : "-null-"), e);
+		}
+		return Optional.empty();
+	}
+
+	public boolean isSpecialNeedsEnabled(Site site) {
+		try {
+			return (serverConfigurationService.getBoolean(SYSTEM_PROP_DISPLAY_SPECIAL_NEEDS, true) || (site != null && Boolean.parseBoolean(site.getProperties().getProperty(SITE_PROP_DISPLAY_SPECIAL_NEEDS))));
+		} catch(Exception e) {
+			log.warn("Error on isSpecialNeedsEnabled (sample) ", e);
```

Comment:
don't use words like Error or Warn as it is redundant and don't use the method name also redundant, simply state the issue.

```suggestion
			log.warn("Could not determine if Special Needs is enabled for site, {}: {}", site, e.toString());
```